### ISSUE TITLE
Fix false positives for single-line comments when --fix is used in max-line-length

### ISSUE THREAD

**Issue by valignatev** [stylelint/stylelint#3958]
Stylelint `v9.10.1` yells at comment lines with the max-line-length, although the line is two characters less than the error threshold. E.g. maximum allowed line length is 80, but the linter fails at 79 already.

> Which rule, if any, is the bug related to?

`max-line-length`

> What CSS is needed to reproduce the bug?

It's happening on SCSS files, I haven't check it on pure CSS.

```scss
// Comment line which is one symbol less than a maximum allowed
.some-styles {
   ...
}
```
Long selectors don't trigger the linter though, so only comments affected.

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "extends": "stylelint-config-suitcss"
}
```

> Which version of stylelint are you using?

It's `9.10.1`. The one that works correctly for me: `9.9.0`. So it must be a regression.

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `stylelint "**/*.css`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

It happens in SCSS files for me, but the comment line is at the top level

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

e.g. "The following warnings were flagged:"

```shell
test.scss
11:81  ✖  Expected line length to be no more than 80 characters   max-line-length
```

**Comment by jeddy3** [comment]
@valignatev Thanks for the report and for using the template.

I'm not able to reproduce this using the demo.

Given the following config (taken from suitcss):

```json
{
  "rules": {
    "max-line-length": [ 80, { "ignore": "non-comments" } ]
  }
}
```

The following doesn't produce a violation:

```scss
// This comment is 80 characters long INCLUDING THE SLASH SLASH + SPACE AT START
```

Adding an extra character does trigger a violation too.

Can you share with us the contents of your `test.scss` file so we can dig a little deeper, please?

**Comment by valignatev** [comment]
Hey @jeddy3, thanks a lot for the fast response and for the awesome tool! It seems that `--syntax scss --fix` flags are needed to trigger false positive. I forgot to mention that because I have figured it out just now, after your comment. Otherwise, your reproduction setup is perfect. This should do:
```shell
stylelint --config .stylelintrc.json test.scss --syntax scss --fix
```

**Comment by jeddy3** [comment]
Thanks for the further info. You're right, `--fix` is needed to reproduce the issue. 

Unfortunately, I've no idea off-hand what could be causing `--fix` to trigger the issue. I'll label up as bug and help wanted.

The only recent change to the rule was done in https://github.com/stylelint/stylelint/pull/3875, we might have introduced a regression there. @gucong3000 Could this be the case?

---

Incidentally, it's unlikely you need to use the `--syntax` option these days. stylelint now uses postcss-syntax, which uses heuristics to automatically select the appropriate syntax.

**Comment by valignatev** [comment]
Thanks for the suggestion! Indeed, omitting `--syntax scss` works fine even with the `--fix` in the `9.10.1`. Since I'm only using `.scss` extension for files, I'm going with that.

**Comment by jeddy3** [comment]
Closing [as stylistic rules are frozen](https://github.com/stylelint/stylelint/issues/5674). The community is welcome to migrate the rule [to a plugin](https://stylelint.io/developer-guide/plugins) and fix this bug.
